internals/manifestworkSync/builders/auth.go

```diff
@@ -0,0 +1,57 @@
+package builders
+
+import (
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	mw1 "open-cluster-management.io/api/work/v1"
+)
+
+// BuildRole creates a mw1.Manifest with a Role for pod log access in a specific namespace.
+func BuildRole(cappName string, namespace string) mw1.Manifest {
+	role := rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cappName + "-logs-reader",
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Resources: []string{
+					"pod/logs",
+				},
+				APIGroups: []string{
+					"",
+				},
+				Verbs: []string{
+					"get", "watch", "list",
+				},
+			},
+		},
+	}
+	return mw1.Manifest{RawExtension: runtime.RawExtension{Object: &role}}
+}
```

Comment:
We can remove the entire Role and RoleBinding stuff since we won't end up needing it, keep it for now but open a separate issue for that so we can remove it before the next release